lib/LogProcessor.js

```diff
@@ -0,0 +1,4 @@
+class LogProcessor {
+}
```

Comment:
What is the role of this base class? I think it can be completely removed.

Reply:
The base class is used in `log-processors.js` in the `createLogProcessor` factory function to quickly identify built-in log processors. I suppose the `instanceof` checks can be replaced with member checks (e.g., `definition.parse || definition.build` or something similar instead, but the current implementation seems more readable.

I actually considered exposing it so that consumers could directly inherit from it, but I was concerned about changing the API.